homeassistant/components/husqvarna_automower/binary_sensor.py

```diff
@@ -0,0 +1,82 @@
+"""Creates a the binary sensor entities for the mower."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+
+from aioautomower.model import MowerActivities, MowerAttributes
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AutomowerDataUpdateCoordinator
+from .entity import AutomowerBaseEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class AutomowerBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes Automower sensor entity."""
+
+    value_fn: Callable[[MowerAttributes], bool]
+
+
+BINARY_SENSOR_TYPES: tuple[AutomowerBinarySensorEntityDescription, ...] = (
+    AutomowerBinarySensorEntityDescription(
+        key="battery_charging",
+        value_fn=lambda data: (data.mower.activity is MowerActivities.CHARGING),
```

Comment:
```suggestion
        value_fn=lambda data: data.mower.activity == MowerActivities.CHARGING,
```
Not sure if there is a big difference here